File: mycollab-core/src/main/java/com/esofthead/mycollab/core/arguments/StringSearchField.java
package com.esofthead.mycollab.core.arguments;

public class StringSearchField extends SearchField {
	private String value;

	public StringSearchField(String value) {
		this(SearchField.AND, value);
	}

	public StringSearchField(String oper, String value) {
		this.operation = oper;
		this.value = value;
	}

	public String getValue() {
		return value;
	}

	public void setValue(String value) {
		this.value = value;
	}
}

File: mycollab-dao/src/main/scala/com/mycollab/db/arguments/StringSearchField.scala
package com.mycollab.db.arguments

import com.mycollab.core.utils.StringUtils

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd
  * @since 5.3.5
  */
class StringSearchField(operation: String, @BeanProperty val value: String) extends SearchField(operation) {
  def this() = this(SearchField.AND, "")
}

object StringSearchField {
  def and(value: String): StringSearchField = if (StringUtils.isNotBlank(value)) new StringSearchField(SearchField.AND, value) else null
}
